pkg/kotsutil/kots.go

```diff
@@ -1598,3 +1600,62 @@ func GetECVersionFromAirgapBundle(airgapBundle string) (string, error) {
 	}
 	return ecVersion, nil
 }
+
+func FindChannelIDInLicense(requestedSlug string, license *kotsv1beta1.License) (string, error) {
+	matchedChannelID := ""
+	if requestedSlug != "" {
+		// if we do not have a Channels array or its empty, default to using the top level fields for backwards compatibility
+		if len(license.Spec.Channels) == 0 {
+			logger.Debug("not a multi-channel license, using top level license channel id")
+			matchedChannelID = license.Spec.ChannelID
+		} else {
+			for _, channel := range license.Spec.Channels {
+				if channel.ChannelSlug == requestedSlug {
+					matchedChannelID = channel.ChannelID
+					break
+				}
+			}
+			if matchedChannelID == "" {
+				return "", errors.New("requested install channel slug not found in license channels")
+			}
+		}
+	} else { // this is an install from before the channel slug was added to the configmap
+		logger.Debug("requested channel slug not found in configmap, using top level channel id from license")
+		matchedChannelID = license.Spec.ChannelID
+	}
+	return matchedChannelID, nil
+}
+
+func FindChannelInLicense(channelID string, license *kotsv1beta1.License) (*kotsv1beta1.Channel, error) {
+	if channelID == "" || len(license.Spec.Channels) == 0 {
+		if license.Spec.ChannelID != channelID {
+			return nil, errors.New("channel not found in license")
```

Comment:
nit: same error in two places in the same function. it will be impossible to know which code path failed.